From e96513cb2a654d134c0ee14b435e968feb586bf6 Mon Sep 17 00:00:00 2001
From: Robby Cai <R63905@freescale.com>
Date: Tue, 10 Sep 2013 16:38:48 +0800
Subject: [PATCH 0851/1072] ENGR00278970 fix build error when build capture
 driver as module

git://git.freescale.com/imx/linux-2.6-imx.git imx_3.10.17_1.0.0_beta
commit c4a03e125d2367f52941eeca666270779037a7b0

fix error for 'make modules':
ENROR: "csi_dmareq_rff_enable"
	[drivers/media/platform/mxc/capture/csi_v4l2_capture.ko] undefined!
ERROR: "csi_dmareq_rff_disable"
	[drivers/media/platform/mxc/capture/csi_v4l2_capture.ko] undefined!
ERROR: "csi_regbase"
	[drivers/media/platform/mxc/capture/csi_v4l2_capture.ko] undefined!
make[1]: *** [__modpost] Error 1
make: *** [modules] Error 2

Signed-off-by: Robby Cai <R63905@freescale.com>
Signed-off-by: Hongbo Zhong <hongbo.zhong@windriver.com>
---
 drivers/media/platform/mxc/capture/fsl_csi.c |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/drivers/media/platform/mxc/capture/fsl_csi.c b/drivers/media/platform/mxc/capture/fsl_csi.c
index 4eb4ac0..4ba59e8 100644
--- a/drivers/media/platform/mxc/capture/fsl_csi.c
+++ b/drivers/media/platform/mxc/capture/fsl_csi.c
@@ -34,6 +34,7 @@
 #include "fsl_csi.h"
 
 void __iomem *csi_regbase;
+EXPORT_SYMBOL(csi_regbase);
 static int irq_nr;
 static csi_irq_callback_t g_callback;
 static void *g_callback_data;
@@ -228,6 +229,7 @@ void csi_dmareq_rff_enable(void)
 	cr3 |= BIT_HRESP_ERR_EN;
 	__raw_writel(cr3, CSI_CSICR3);
 }
+EXPORT_SYMBOL(csi_dmareq_rff_enable);
 
 void csi_dmareq_rff_disable(void)
 {
@@ -237,6 +239,7 @@ void csi_dmareq_rff_disable(void)
 	cr3 &= ~BIT_HRESP_ERR_EN;
 	__raw_writel(cr3, CSI_CSICR3);
 }
+EXPORT_SYMBOL(csi_dmareq_rff_disable);
 
 static const struct of_device_id fsl_csi_dt_ids[] = {
 	{ .compatible = "fsl,imx6sl-csi", },
-- 
1.7.5.4

